csharp/src/Ice/Properties.cs

```diff
@@ -324,6 +334,12 @@ public void setProperty(string key, string value)
         // Check if the property is in an Ice property prefix. If so, check that it's a valid property.
         if (findIcePropertyArray(key) is PropertyArray propertyArray)
         {
+            if (propertyArray.isOptIn && !_optInPrefixes.Contains(propertyArray.name))
+            {
+                throw new PropertyException($"unable to set '{key}': property prefix '{propertyArray.name}' is " +
+                "opt-in and must be explicitly enabled");
```

Comment:
In C# we prefer the message in a single line even if exceed the max line length
```suggestion
                throw new PropertyException(
                    $"unable to set '{key}': property prefix '{propertyArray.name}' is opt-in and must be explicitly enabled");
```